File: typescript/02/star_one.ts
import * as fs from "fs/promises";
import * as path from "path";

const INVENTORY = {
    red: 12,
    green: 13,
    blue: 14
}

function gameIsPossible(input: string): boolean {
    for (const s of input.split("; ")) {
        const countNum = parseInt(s.split(" ")[0])
        const cube = s.split(" ")[1]
        if (countNum > INVENTORY[cube]) return false
    }
    return true
}

(async () => {
    const rawInput = await fs.readFile(path.resolve(__dirname, '../../input/day_02.txt'), "utf-8");
    const arr = rawInput.split("\n");

    const total = arr.reduce((sum, line) => {
        const gameNum = parseInt(line.split(": ")[0].substring(5))
        const input = line.split(": ")[1].replaceAll(",", ";")

        return sum + ( gameIsPossible(input) ? gameNum : 0 );
    }, 0);

    console.log(total);
})();

File: python/02/star_one.py
import os

INVENTORY = {
    "red": 12,
    "green": 13,
    "blue": 14
}

input_file = os.path.join(os.path.dirname(__file__), "../../input/day_02.txt")

def game_is_possible(input: str) -> bool:
    for s in input.replace(",", ";").split("; "):
        count_str, cube = s.split(" ")
        if int(count_str) > INVENTORY[cube]:
            return False
    return True

total = 0
with open(input_file, "r") as file:
    for x in file.readlines():
        x = x[:-1]  # remove \n
        game_str, input = x.split(": ")
        if game_is_possible(input):
            total += int(game_str[5:])  # remove 'Game '
print(total)